pkg/controlplane/apiserver/config.go

```diff
@@ -124,6 +124,7 @@ func BuildGenericConfig(
 
 	kubeVersion := version.Get()
 	genericConfig.Version = &kubeVersion
```

Comment:
would it make sense to replace the Version field with the EffectiveVersion field to flush out anything relying on Version and push them to use one of the EffectiveVersion getters